plugins/modules/aci_cloud_cidr.py

```diff
@@ -0,0 +1,289 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2020, nkatarmal-crest <nirav.katarmal@crestdatasys.com>
+# Copyright: (c) 2020, Cindy Zhao <cizhao@cisco.com>
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+DOCUMENTATION = r'''
+---
+module: aci_cloud_cidr
+short_description: Manage CIDR under Cloud Context Profile (cloud:Cidr)
+description:
+-  Manage Cloud CIDR on Cisco Cloud ACI.
+notes:
+- More information about the internal APIC class B(cloud:Cidr) from
+  L(the APIC Management Information Model reference,https://developer.cisco.com/docs/apic-mim-ref/).
+- This module is only used to manage non_primary Cloud Cidr, the primary Cidr is created by aci_cloud_ctx_profile module.
+author:
+- Nirav (@nirav)
+- Cindy Zhao (@cizhao)
+options:
+  address:
+    description:
+    - CIDR ip and its submask.
+    type: str
+  description:
+    description:
+    - Description of the Cloud CIDR.
+    type: str
+  name_alias:
+    description:
+    - An alias for the name of the current object. This relates to the nameAlias field in ACI and is used to rename object without changing the DN.
+    type: str
+  tenant:
+    description:
+    - The name of the Tenant.
+    type: str
+  cloud_context_profile:
+    description:
+    - The name of the Cloud Context Profile.
+    type: str
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    choices: [ absent, present, query ]
+    default: present
+    type: str
+
+extends_documentation_fragment:
+- cisco.aci.aci
+'''
+
+EXAMPLES = r'''
+- name: Create non_primary cidr
+  aci_cloud_cidr:
+    host: apic
+    username: admin
+    password: SomeSecretPassword
+    tenant: tenantName
+    address: 10.10.10.1/16
```

Comment:
10.10.10.1/16 is not a valid subnet -> 10.10.0.0/16 (please also fix the following ones)